src/thunderbots/software/ai/world/team.cpp

```diff
@@ -61,6 +61,12 @@ void Team::removeExpiredRobots(const Timestamp& timestamp)
     // has passed, then remove the robot from the team
     for (auto it = team_robots.begin(); it != team_robots.end();)
     {
+        Duration time_diff = timestamp - it->second.lastUpdateTimestamp();
+        if (time_diff.getSeconds() < 0)
+        {
+            throw std::invalid_argument(
+                "Error: tried to remove a robot at a negative time");
+        }
         if ((timestamp - it->second.lastUpdateTimestamp()) > robot_expiry_buffer_duration)
```

Comment:
Can we replace `(timestamp - it->second.lastUpdateTimestamp())` with the new `time_diff` here?

Reply:
absolutely, changed